api/src/api/userRouter.ts

```diff
@@ -321,20 +324,50 @@ export const userRouterFactory = (
   };
 
   // eslint-disable-next-line unicorn/consistent-function-scoping
-  const asyncUpdateAndSaveLicenseStatusIfNeeded = async (userData: UserData): Promise<void> => {
-    const currentBusinessLicenseData = getCurrentBusiness(userData).licenseData;
-    if (!currentBusinessLicenseData || !shouldCheckLicense()) {
-      return;
-    }
+  const asyncUpdateAndSaveLicenseStatusIfNeeded = async (userData: UserData): Promise<UserData> => {
+    const currentBusiness = getCurrentBusiness(userData);
+
+    if (!shouldCheckLicense(currentBusiness)) return userData;
+
+    try {
+      let nameAndAddress: LicenseSearchNameAndAddress;
+      const useLicenseAddress =
+        currentBusiness.licenseData !== undefined && currentBusiness.licenseData.licenses !== undefined;
+      if (useLicenseAddress) {
+        const licenses = currentBusiness.licenseData.licenses;
+        const firstLicenseName = Object.keys(licenses!)[0] as LicenseName;
+        const licenseNameAndAddress = licenses![firstLicenseName]?.nameAndAddress;
```

Comment:
Agreed. I had done this because TypeScript wasn't smart enough to know that I'm checking `licenses` is defined as a part of `useLicenseAddress`.

I was thinking I was making it more explicit, but I may have inadvertently made the code more clunky.

Reply:
Updated